core/src/main/scala/kanaloa/reactive/dispatcher/queue/Queue.scala

```diff
@@ -58,9 +59,7 @@ trait Queue extends Actor with ActorLogging with MessageScheduler {
       finish(status, s"Queue successfully retired")
       PartialFunction.empty //doesn't matter after finish, but is required by the api.
     } else handleWork(status, retiring) orElse {
-      case Enqueue(_, replyTo, _) ⇒
-        replyTo.getOrElse(sender) ! Retiring
-
+      case Enqueue         ⇒ sender() ! Retiring
```

Comment:
This is not going to match, you are trying to match the class. Since the tests didn't catch this, do you mind add one?